dvc/repo/experiments/remove.py

```diff
@@ -67,14 +67,21 @@ def _get_exp_stash_index(repo, ref_or_rev: str) -> Optional[int]:
     return None
 
 
-def _get_exp_ref(repo, remote: str, exp_name: str) -> Optional[ExpRefInfo]:
+def _get_exp_ref(
+    repo, remote: Optional[str], exp_name: str
+) -> Optional[ExpRefInfo]:
+    if exp_name.startswith(EXPS_NAMESPACE):
+        if repo.scm.get_ref(exp_name):
```

Comment:
I don't think this check is actually necessary, we can just return the refinfo directly. Also, we can just check if it starts with `refs/` and not `refs/exps`. 

Essentially, if the user gives us a non-exp `refs/...` string, it won't match anything in the rest of this function. Likewise, if `get_ref()` fails, it also means that nothing would match in the rest of this function. So it's faster for us to just return the result of `ExpRefInfo.from_ref()` immediately.